### ISSUE TITLE
Wrong output

### ISSUE THREAD

**Issue by lbineau** [NaturalIntelligence/fast-xml-parser#1]
Hi,
Thank you for your library.
I have a problem with this XML that is transformed into the following JSON me this and the structure is really not the same between the two.

```xml
<s:Envelope xmlns:s="http://www.w3.org/2003/05/soap-envelope" xmlns:a="http://www.w3.org/2005/08/addressing">
    <s:Header>
        <a:Action s:mustUnderstand="1">http://www.b-i.com/ns/LoyaltyWebservice/IParticipationService/GetUserProfileResponse</a:Action>
    </s:Header>
    <s:Body>
        <GetUserProfileResponse xmlns="http://www.b-i.com/ns/LoyaltyWebservice">
            <GetUserProfileResult xmlns:i="http://www.w3.org/2001/XMLSchema-instance">
                <Data i:type="ConsumerData">
                    <UserProfile>
                        <Age>28</Age>
                        <Birthday>1988-07-13T00:00:00</Birthday><BrandComPrefID i:nil="true"/><BrandComPrefName i:nil="true"/>
                        <City>Annecy</City>
                        <ConsumerID>5081695</ConsumerID>
                        <ConsumerNumber>5081695</ConsumerNumber><DeeplinkToken i:nil="true"/>
                        <DeliveryCity>Annecy</DeliveryCity><DeliveryPhone i:nil="true"/>
                        <DeliveryStreetAddress>avenue de geneve</DeliveryStreetAddress><DeliveryStreetNumber i:nil="true"/>
                        <DeliveryZipCode>74000</DeliveryZipCode>
                        <Email>lory.bineau@b-i.com</Email>
                        <FestivalId>0</FestivalId>
                        <FirstName>Lory</FirstName>
                        <Gender>Male</Gender><HomePhone i:nil="true"/>
                        <IsLAMP>false</IsLAMP>
                        <IsMainBrandPM>false</IsMainBrandPM>
                        <IsOccasionalBrandPM>false</IsOccasionalBrandPM>
                        <LastName>Bineau</LastName>
                        <LoginID>1250221</LoginID><LoginNumber i:nil="true"/>
                        <MainBrandID>8907</MainBrandID><MainBrandName i:nil="true"/>
                        <MainBrandOpinion>0</MainBrandOpinion><MobilePhone i:nil="true"/><OccasionalBrandID i:nil="true"/><OccasionalBrandName i:nil="true"/><POBoxCity i:nil="true"/><POBoxNumber i:nil="true"/><POBoxZIPCode i:nil="true"/>
                        <PersonId>5081695</PersonId>
                        <PreferredLanguage>2</PreferredLanguage>
                        <Properties>
                            <Properties>
                                <ExtendedProperty>
                                    <Key>CanClaimIncentive</Key>
                                    <Value>False</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>Competitor</Key>
                                    <Value>False</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>FullFlavor</Key>
                                    <Value>True</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>Segment</Key>
                                    <Value>0</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>Above8</Key>
                                    <Value>True</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>GoldSecondary</Key>
                                    <Value>False</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>SpokenLanguage</Key>
                                    <Value>FR</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>ShowGame</Key>
                                    <Value>False</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>PointBalance</Key>
                                    <Value>50</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>CarTickets</Key>
                                    <Value>0</Value>
                                </ExtendedProperty>
                                <ExtendedProperty>
                                    <Key>Influencer</Key>
                                    <Value>False</Value>
                                </ExtendedProperty>
                            </Properties>
                        </Properties>
                        <StreetAddress>avenue de geneve</StreetAddress>
                        <StreetNumber>43b</StreetNumber>
                        <WebSurveyCompleted>false</WebSurveyCompleted>
                        <ZipCode>74000</ZipCode>
                    </UserProfile>
                </Data>
                <Details>Lory Bineau</Details>
                <Result>200</Result>
            </GetUserProfileResult>
        </GetUserProfileResponse>
    </s:Body>
</s:Envelope>
```
=>
```json
{
    "s:Envelope": {
        "s:Header": {
            "a:Action": "http://www.b-i.com/ns/LoyaltyWebservice/IParticipationService/GetUserProfileResponse"
        },
        "s:Body": {
            "GetUserProfileResponse": {
                "GetUserProfileResult": {
                    "Data": {
                        "UserProfile": {
                            "Age": 28,
                            "Birthday": "1988-07-13T00:00:00",
                            "BrandComPrefID": {
                                "BrandComPrefName": {
                                    "City": "Annecy",
                                    "ConsumerID": 5081695,
                                    "ConsumerNumber": 5081695,
                                    "DeeplinkToken": {
                                        "DeliveryCity": "Annecy",
                                        "DeliveryPhone": {
                                            "DeliveryStreetAddress": "avenue de geneve",
                                            "DeliveryStreetNumber": {
                                                "DeliveryZipCode": 74000,
                                                "Email": "lory.bineau@b-i.com",
                                                "FestivalId": {},
                                                "FirstName": "Lory",
                                                "Gender": "Male",
                                                "HomePhone": {
                                                    "IsLAMP": "false",
                                                    "IsMainBrandPM": "false",
                                                    "IsOccasionalBrandPM": "false",
                                                    "LastName": "Bineau",
                                                    "LoginID": 1250221,
                                                    "LoginNumber": {
                                                        "MainBrandID": 8907,
                                                        "MainBrandName": {
                                                            "MainBrandOpinion": {},
                                                            "MobilePhone": {
                                                                "OccasionalBrandID": {
                                                                    "OccasionalBrandName": {
                                                                        "POBoxCity": {
                                                                            "POBoxNumber": {
                                                                                "POBoxZIPCode": {
                                                                                    "PersonId": 5081695,
                                                                                    "PreferredLanguage": 2,
                                                                                    "Properties": {
                                                                                        "Properties": {
                                                                                            "ExtendedProperty": [{
                                                                                                "Key": "CanClaimIncentive",
                                                                                                "Value": "False"
                                                                                            }, {
                                                                                                "Key": "Competitor",
                                                                                                "Value": "False"
                                                                                            }, {
                                                                                                "Key": "FullFlavor",
                                                                                                "Value": "True"
                                                                                            }, {
                                                                                                "Key": "Segment",
                                                                                                "Value": {}
                                                                                            }, {
                                                                                                "Key": "Above8",
                                                                                                "Value": "True"
                                                                                            }, {
                                                                                                "Key": "GoldSecondary",
                                                                                                "Value": "False"
                                                                                            }, {
                                                                                                "Key": "SpokenLanguage",
                                                                                                "Value": "FR"
                                                                                            }, {
                                                                                                "Key": "ShowGame",
                                                                                                "Value": "False"
                                                                                            }, {
                                                                                                "Key": "PointBalance",
                                                                                                "Value": 50
                                                                                            }, {
                                                                                                "Key": "CarTickets",
                                                                                                "Value": {}
                                                                                            }, {
                                                                                                "Key": "Influencer",
                                                                                                "Value": "False"
                                                                                            }]
                                                                                        }
                                                                                    },
                                                                                    "StreetAddress": "avenue de geneve",
                                                                                    "StreetNumber": "43b",
                                                                                    "WebSurveyCompleted": "false",
                                                                                    "ZipCode": 74000
                                                                                }
                                                                            },
                                                                            "Details": "Lory Bineau",
                                                                            "Result": 200
                                                                        }
                                                                    }
                                                                }
                                                            }
                                                        }
                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

**Comment by amitguptagwl** [comment]
Thanks for the information. I'll have a look in the evening.

Note: Currently I skip arguments from parsing. I'm planning them to make as a part of configuration.

**Comment by amitguptagwl** [comment]
It seems that the issue is because of the self closing tags present in the XML. Can you plz tell me what  output are you expecting from the above XML?

I'm just wondering if the value of self closing tag should be set to blank as I ignore the attributes of any tag.

**Comment by amitguptagwl** [comment]
I have released another version to address above issue and with few new features. But it is failing for above XML. So I'll fix that issue in the evening today and will update you.

**Comment by amitguptagwl** [comment]
Just released 2.1.0. Plz have a look on read me as well.
